Audio PR7 - Audio augmentations 2 (MFBB and Normalize)

SundarRajan28 opened the pull request.

SundarRajan28 committed:
Merge branch 'swbs/audio/pr9' into swbs/audio/pr7

swetha097 committed:
Minor Change

swetha097 committed:
Fix C++ audio unit test for 2D data

SundarRajan28 committed:
Minor changes

swetha097 committed:
Minor fix in verify output

SundarRajan28 committed:
Change in API docs

sampath1117 reviewed (commented):
Please address review comments

SundarRajan28 committed:
Merge branch 'swbs/audio/pr9' into swbs/audio/pr7

SundarRajan28 committed:
Merge branch 'swbs/audio/pr9' into swbs/audio/pr7

SundarRajan28 committed:
Merge branch 'swbs/audio/pr9' into swbs/audio/pr7

SundarRajan28 committed:
Merge branch 'swbs/audio/pr9' into swbs/audio/pr7

SundarRajan28 committed:
Merge branch 'swbs/audio/pr8' into swbs/audio/pr7

SundarRajan28 committed:
Adding changes to MFB and normalize nodes

fiona-gladwin committed:
Merge branch 'swbs/audio/pr9' into swbs/audio/pr7

swetha097 committed:
Merge branch 'swbs/audio/pr9' into swbs/audio/pr7

SundarRajan28 committed:
Merge remote-tracking branch 'upstream/develop' into swbs/audio/pr7

SundarRajan28 committed:
Merge remote-tracking branch 'upstream/develop' into swbs/audio/pr7

SundarRajan28 committed:
Fix merge conflicts

SundarRajan28 committed:
Merge remote-tracking branch 'upstream/develop' into swbs/audio/pr7

SundarRajan28 committed:
Merge remote-tracking branch 'upstream/develop' into swbs/audio/pr7

SundarRajan28 committed:
Resolving review comments

SundarRajan28 committed:
Merge branch 'swbs/audio/pr9' into swbs/audio/pr7

kiritigowda committed:
CentOS 7 - Support (#170)

* Setup - CentOS 7 Updates

* Setup - Fix

* CMakeLists - CentOS 7 Package updates

* CMake3 Install on setup

* Setup - Updates

* Updates - Readme & JPEG

jayhawk-commits committed:
CI - Enable external CI pipeline triggers (#171)

kiritigowda committed:
SLES 15 SP5 - support updates (#172)

* SLES - Package updates

* SLES - Updates

* Setup - Updates

* rocAL Setup - Cleanup

* SLES - OpenCV Package

swetha097 committed:
Audio PR - Augmentation support [ NonSilentRegionDetection and Slice ] (#128)

* Modifying python unittests

* Fixing spectogram test case

* Remove the reset_tensor_roi calls

* Resolving review comments

* Resolve some PR comments

* Minor changes

* Resolving review comments

* Change the dims[0] and dims[1] positioning for Spectrogram

* Resolving review comments

* Resolving review comments

* Resolving PR comments

* Updating audio unit tests for default file list path

* Minor changes

* Minor Change

* Minor change

* Name change from sample to data

* Change from decoded_data_info to DecodedDataInfo

* Revert "Change the dims[0] and dims[1] positioning for Spectrogram"

This reverts commit d791b9ac77d662f1528088165d2cfdba721a2b4f.

* Remove audio_decoder_factory.cpp file

* Minor change

* Change variable name

* Change copyright year and remove debug statements

* Add Spectrogram Case in unit tests

* Add spectrogram case in python unit tests

* Update the struct variable name in audio files

* Fixing issues with downmix node output

* Adding ROI updation in downmix node

* Adding downmix test case for python unit tests

* Adding downmix and to_decibels test case in C++ tests

* Minor changes

* Change ROCAL_DATA_PATH to exclude rocal_data

* Update ROCAL_DATA_PATH to exclude rocal_data

* Use Pascal case for function names in audio decoder

* Adding mel filter bank augmentation support in rocAL

* Add resample test case and fixed issues

* NWC - NSR + Slice

* Minor changes

* Minor changes

* Adding latest changes to resample and distribution nodes

* Add test suite changes for NSR

* Fixing ROI handle errors with distribution nodes

* Resolve issue with Slice Node

* Add audio path for downmix test case

* Add NSR and Slice in python unit tests

* Resolving review comments

* Add resample and operator overloading test cases in test suite

* Resolving review comments

* Minor change

* Resolving review comments

* Fix review comments

* Modify cmake to have SNDFILE in all capital

* Fix Slice

* Add NSR and slice in audio test suite

* Fix NSR and Slice output issue

* Resolving review comments

* Adding set_seed in master_graph ctor

* Change Name NonSilentRegion to NonSilentRegionDetection

* fix output mismatch for Tensor add Tensor

* Remove nD tensor layouts

* Minor changes

* Add struct for audio info in AudioReadAndDecode

* Fix merge conflict

* Clean up - node_slice.cpp

* Resolve PR comments - 1

* Remove unused variables, add comments, format merge commit

* Resolve internal PR comments

* Changes in node_slice.cpp

* Change Enum fom UINT TO INT type

* Minor change

* Fornat

* Fix pybind issue for normalize

* Changes in node_slice.cpp to move the fill_values to create_node from update_node

* Add _vx to scalar variables in the nsr

* Moving normalize to effects augmentations folder

* Renaming crop_image_info to CropImageInfo

* Remove - actual_host_buffers - Unused

* Rename TimingDBG to TimingDbg

* Move the instances of DecodedDataInfo to its base class LoaderModule

* Fix a WRN msg in master_graph.cpp

* Remove a dangling comment

* Rename _circ_data_info to _circ_buff_data_info

* Adding changes to normalize node

* Add Glob to CMakeLists.txt

* Rename SndFileDecoder to GenericAudioDecoder

* Fix build issues

* Minor change

* Update python API README.md for audio unit test

* Update audio unit test README

* Add debug and changes to normalize node and test case

* Minor fixes for MFB

* Add Normalize and MFB test cases in C++ tests

* Revert "Merge remote-tracking branch 'swe_fork/swbs/audio/pr7' into swbs/audio/pr9"

This reverts commit e480f982957e644d0b452dc372ecfcf5352e899b, reversing
changes made to e5e9ad2e07fbfd4625c08d1e4a7663776051c23d.

* Fix cpp unit test

* Adding missed param in python unit tests

* Revert "Add Glob to CMakeLists.txt"

This reverts commit 47263d94393e9614a5cebcab6c11c804ac7152f8.

* Fix include headers for Audio files

* Fix copy data 2D

* Minor changes

* Pass decoded data info to load routine instead of separate vectors

* Update CHANGELOG.md

* Update CHANGELOG.md

* Change swap_handle_time variable name in loader

* Update the changelog.md

* Update ChangeLog.md

* Update ChangeLog.md

* Update CHANGELOG.md

* Formatting changes

Add comments

* Update doxygen comments

* Move file source reader from readers/image to readers folder

* Update README and add doxygen description

* Update CMakeLists and README for audio test

* Update README for audio test

* Minor fix

* Fix merge from PR 2

* Minor changes shard_count argument name

* Rename set and get functions of data_info to decoded_data_info

* Revert empty line removed in CMakeLists.txt

* Removed prefix original for audio vectors

* Resolve PR comments

* Add @params to all args in pytorch.py

* Fix build issue

* Minor changes in unit test

* Minor changes

* Change ROCAL instaces to rocAL in pytorch.py

* Resolve the PR comments

* Minor changes in decoders.py - Modify the comment for shard_size

* Minor changes

* Address the PR comments

* Address Review comments

* Introduce Audio layouts

* Add layout changes for spectrogram

* Fix the unit tests - c++ & python

* Minor fix

* Adding changes for spec layout changes

* Update node_slice.cpp

* Update node_slice.h

* Resolving review comments

---------

Co-authored-by: SundarRajan28 <sundarrajan@multicorewareinc.com>
Co-authored-by: Swetha B S <>
Co-authored-by: Fiona-MCW <70996026+fiona-gladwin@users.noreply.github.com>
Co-authored-by: fiona-gladwin <fionagladwin@multicorewareinc.com>
Co-authored-by: Lakshmi Kumar <lakshmi.kumar@amd.com>

dependabot[bot] committed:
Docs - Bump rocm-docs-core[api_reference] from 1.4.0 to 1.4.1 in /docs/sphinx (#173)

Bumps [rocm-docs-core[api_reference]](https://github.com/ROCm/rocm-docs-core) from 1.4.0 to 1.4.1.
- [Release notes](https://github.com/ROCm/rocm-docs-core/releases)
- [Changelog](https://github.com/ROCm/rocm-docs-core/blob/develop/CHANGELOG.md)
- [Commits](https://github.com/ROCm/rocm-docs-core/compare/v1.4.0...v1.4.1)

---
updated-dependencies:
- dependency-name: rocm-docs-core[api_reference]
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Kiriti Gowda <kiritigowda@gmail.com>

SundarRajan28 committed:
Merge remote-tracking branch 'upstream/develop' into swbs/audio/pr7

kiritigowda committed:
CMakeLists - AMDGPU_TARGETS Updates (#174)

LakshmiKumar23 committed:
Merge branch 'develop' into swbs/audio/pr7

SundarRajan28 committed:
Resolving review comments

dependabot[bot] committed:
Bump rocm-docs-core[api_reference] from 1.4.1 to 1.5.0 in /docs/sphinx (#178)

Bumps [rocm-docs-core[api_reference]](https://github.com/ROCm/rocm-docs-core) from 1.4.1 to 1.5.0.
- [Release notes](https://github.com/ROCm/rocm-docs-core/releases)
- [Changelog](https://github.com/ROCm/rocm-docs-core/blob/develop/CHANGELOG.md)
- [Commits](https://github.com/ROCm/rocm-docs-core/compare/v1.4.1...v1.5.0)

---
updated-dependencies:
- dependency-name: rocm-docs-core[api_reference]
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

LakshmiKumar23 committed:
Merge branch 'develop' into swbs/audio/pr7

SundarRajan28 committed:
Resolving review comments

samjwu committed:
Docs - Update documentation link (#175)

Co-authored-by: Kiriti Gowda <kiritigowda@gmail.com>

kiritigowda committed:
Merge branch 'develop' into swbs/audio/pr7
